app/src/main/java/com/gh4a/BaseActivity.java

```diff
@@ -586,9 +609,19 @@ private void setupSwipeToRefresh() {
                     UiUtils.resolveColor(this, R.attr.colorPrimaryDark), 0
             );
         }
+
+        mSwipeLayoutBehavior = new BottomSheetFixScrollingViewBehavior();
+        CoordinatorLayout.LayoutParams lp =
+                (CoordinatorLayout.LayoutParams) mSwipeLayout.getLayoutParams();
+        lp.setBehavior(mSwipeLayoutBehavior);
+
         updateSwipeToRefreshState();
     }
 
+    public void setBottomSheetFixEnabled(boolean enabled) {
```

Comment:
This is about the only part of the whole diff that I'm unhappy with, because this method is a bit unclear: from reading the code, it's not obvious what the 'bottom sheet fix' is and why it's needed.

I think I'd prefer a method that instantiates the swipe layout behavior instead, returning a normal ScrollingViewBehavior by default and a BottomSheetFixScrollingViewBehavior (BTW, name suggestion: 'BottomSheetCompatibleScrollingViewBehavior'?) for the activities that host the bottom sheet fragment.